File: src/main/java/fi/vm/sade/lokalisointi/model/LocalisationOverride.java
package fi.vm.sade.lokalisointi.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.PersistenceCreator;
import org.springframework.data.relational.core.mapping.Column;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor(access = AccessLevel.PRIVATE, onConstructor = @__(@PersistenceCreator))
public class LocalisationOverride {
  private @Id @With Integer id;
  private String namespace;
  @NonNull private String locale;

  @Column("localisation_key")
  @NonNull
  private String key;

  @Column("localisation_value")
  @NonNull
  private String value;

  @NonNull private String createdBy;
  @NonNull private LocalDateTime created;
  @NonNull private String updatedBy;
  @NonNull private LocalDateTime updated;

  public LocalisationOverride() {}

  public LocalisationOverride(
      final String namespace,
      @NonNull final String locale,
      @NonNull final String key,
      @NonNull final String value,
      @NonNull final String createdBy,
      @NonNull final String updatedBy) {
    this.namespace = namespace;
    this.locale = locale;
    this.key = key;
    this.value = value;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
    this.created = LocalDateTime.now();
    this.updated = LocalDateTime.now();
  }

  public Localisation toLocalisation() {
    return new Localisation(this.id, this.namespace, this.key, this.locale, this.value);
  }
}

File: src/main/java/fi/vm/sade/lokalisointi/model/LocalisationOverride.kt
package fi.vm.sade.lokalisointi.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import java.time.LocalDateTime

data class LocalisationOverride(
    @Id
    var id: Int? = null,
    var namespace: String,
    var locale: String,
    @Column("localisation_key")
    var key: String,
    @Column("localisation_value")
    var value: String,
    var createdBy: String,
    var created: LocalDateTime,
    var updatedBy: String,
    var updated: LocalDateTime
) {
    fun toLocalisation(): Localisation {
        return Localisation(this.id, this.namespace, this.key, this.locale, this.value)
    }
}
